tests/testthat/test-renderer4-geoms-zoom.R

```diff
@@ -0,0 +1,97 @@
+info$html_updated1 <- getHTML()
+
+# Update selection and get HTML
+clickID(c("plot_g_cyl_variable_4"))
+Sys.sleep(0.5)
+info$html_updated2 <- getHTML()
+
+## Get ranges of geoms
+no_updates_nodes1 <-
+  getNodeSet(info$html_updated1, '//g[@class="geom1_point_g"]//circle')
+no_updates_nodes2 <-
+  getNodeSet(info$html_updated2, '//g[@class="geom1_point_g"]//circle')
+no_updates_attrs1 <- sapply(no_updates_nodes1, xmlAttrs)[c("cx", "cy"), ]
+no_updates_attrs2 <- sapply(no_updates_nodes2, xmlAttrs)[c("cx", "cy"), ]
+no_updates_xranges1 <- range(as.numeric(no_updates_attrs1[1, ]), na.rm = T)
+no_updates_xranges2 <- range(as.numeric(no_updates_attrs2[1, ]), na.rm = T)
+no_updates_yranges1 <- range(as.numeric(no_updates_attrs1[2, ]), na.rm = T)
+no_updates_yranges2 <- range(as.numeric(no_updates_attrs2[2, ]), na.rm = T)
+
+x_updates_nodes1 <-
+  getNodeSet(info$html_updated1, '//g[@class="geom2_point_g1"]//circle')
+x_updates_nodes2 <-
+  getNodeSet(info$html_updated2, '//g[@class="geom2_point_g1"]//circle')
+x_updates_attrs1 <- sapply(x_updates_nodes1, xmlAttrs)[c("cx", "cy"), ]
+x_updates_attrs2 <- sapply(x_updates_nodes2, xmlAttrs)[c("cx", "cy"), ]
+x_updates_xranges1 <- range(as.numeric(x_updates_attrs1[1, ]), na.rm = T)
+x_updates_xranges2 <- range(as.numeric(x_updates_attrs2[1, ]), na.rm = T)
+x_updates_yranges1 <- range(as.numeric(x_updates_attrs1[2, ]), na.rm = T)
+x_updates_yranges2 <- range(as.numeric(x_updates_attrs2[2, ]), na.rm = T)
+
+y_updates_nodes1 <-
+  getNodeSet(info$html_updated1, '//g[@class="geom3_point_g2"]//circle')
+y_updates_nodes2 <-
+  getNodeSet(info$html_updated2, '//g[@class="geom3_point_g2"]//circle')
+y_updates_attrs1 <- sapply(y_updates_nodes1, xmlAttrs)[c("cx", "cy"), ]
+y_updates_attrs2 <- sapply(y_updates_nodes2, xmlAttrs)[c("cx", "cy"), ]
+y_updates_xranges1 <- range(as.numeric(y_updates_attrs1[1, ]), na.rm = T)
+y_updates_xranges2 <- range(as.numeric(y_updates_attrs2[1, ]), na.rm = T)
+y_updates_yranges1 <- range(as.numeric(y_updates_attrs1[2, ]), na.rm = T)
+y_updates_yranges2 <- range(as.numeric(y_updates_attrs2[2, ]), na.rm = T)
+
+xy_updates_nodes1 <-
+  getNodeSet(info$html_updated1, '//g[@class="geom4_point_g3"]//circle')
+xy_updates_nodes2 <-
+  getNodeSet(info$html_updated2, '//g[@class="geom4_point_g3"]//circle')
+xy_updates_attrs1 <- sapply(xy_updates_nodes1, xmlAttrs)[c("cx", "cy"), ]
+xy_updates_attrs2 <- sapply(xy_updates_nodes2, xmlAttrs)[c("cx", "cy"), ]
+xy_updates_xranges1 <- range(as.numeric(xy_updates_attrs1[1, ]), na.rm = T)
+xy_updates_xranges2 <- range(as.numeric(xy_updates_attrs2[1, ]), na.rm = T)
+xy_updates_yranges1 <- range(as.numeric(xy_updates_attrs1[2, ]), na.rm = T)
+xy_updates_yranges2 <- range(as.numeric(xy_updates_attrs2[2, ]), na.rm = T)
+
+
+test_that("geoms get zoomed-in upon changing selection", {
+  # no_updates
+  expect_false(isTRUE(all.equal(no_updates_xranges2, no_updates_xranges1)))
+  expect_false(isTRUE(all.equal(no_updates_yranges2, no_updates_yranges1)))
```

Comment:
to clarify your intent, could you please define `expect_not_equal` in helper-functions.R?